ui/lib/core/addon/components/ttl-form.js

```diff
@@ -2,102 +2,152 @@
  * @module TtlForm
  * TtlForm components are used to enter a Time To Live (TTL) input.
  * This component does not include a label and is designed to take
- * a time and unit, and pass an object including seconds and
- * timestring when those two values are changed.
+ * a time and unit, and pass an object including seconds,
+ * timestring, and go-safe timestring when either values are changed.
  *
  * @example
  * ```js
- * <TtlForm @onChange={{action handleChange}} @unit="m"/>
+ * <TtlForm @onChange={{this.handleChange}} @initialValue="30m"/>
  * ```
  * @param {function} onChange - This function will be called when the user changes the value. An object will be passed in as a parameter with values seconds{number}, timeString{string}
- * @param {number} [time] - Time is the value that will be passed into the value input. Can be null/undefined to start if input is required.
- * @param {unit} [unit="s"] - This is the unit key which will show by default on the form. Can be one of `s` (seconds), `m` (minutes), `h` (hours), `d` (days)
- * @param {number} [recalculationTimeout=5000] - This is the time, in milliseconds, that `recalculateSeconds` will be be true after time is updated
+ * @param {string} [initialValue] - InitialValue is the duration value which will be shown when the component is loaded. If it can't be parsed, will default to 0.
  */
 
 import Ember from 'ember';
-import Component from '@ember/component';
-import { computed } from '@ember/object';
-import { task, timeout } from 'ember-concurrency';
-import layout from '../templates/components/ttl-form';
+import Component from '@glimmer/component';
+import { restartableTask, timeout } from 'ember-concurrency';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import Duration from '@icholy/duration';
 
-const secondsMap = {
+export const secondsMap = {
   s: 1,
   m: 60,
   h: 3600,
   d: 86400,
 };
-const convertToSeconds = (time, unit) => {
+export const convertToSeconds = (time, unit) => {
   return time * secondsMap[unit];
 };
-const convertFromSeconds = (seconds, unit) => {
+export const convertFromSeconds = (seconds, unit) => {
   return seconds / secondsMap[unit];
 };
+export const goSafeConvertFromSeconds = (seconds, unit) => {
+  // Go only accepts s, m, or h units
+  const u = unit === 'd' ? 'h' : unit;
+  return convertFromSeconds(seconds, u) + u;
+};
+const largestUnitFromSeconds = (seconds) => {
+  let unit = 's';
+  // get largest unit with no remainder
+  if (seconds % secondsMap.d === 0) {
+    unit = 'd';
+  } else if (seconds % secondsMap.h === 0) {
+    unit = 'h';
+  } else if (seconds % secondsMap.m === 0) {
+    unit = 'm';
+  }
+  return unit;
+};
 
-export default Component.extend({
-  layout,
-  time: '',
-  unit: 's',
+export default class TtlFormComponent extends Component {
```

Comment:
This component looks almost the same as `TtlPicker2`. I think this corresponds to our conversation on Slack but now that I'm seeing the code I'm wondering if we can just conditionally show the toggle button and get rid of this component?

Reply:
Or perhaps this component could extend `TtlPicker2` and we can add whatever is specifically needed for this iteration?

Reply:
My intention was to separate them so that the enable logic was separate from the TTL picker logic (which is technically 2 inputs). However, they are so tightly integrated that maybe just having one component makes the most sense (and we keep `@hideToggle`

Reply:
Went back and consolidated the two into one ttl-picker component!